assets/js/modules/analytics-4/datastore/audience-settings.js

```diff
@@ -0,0 +1,278 @@
+/**
+ * `modules/analytics-4` data store: audience settings.
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import { isPlainObject } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { commonActions, createReducer } from 'googlesitekit-data';
+import { MODULES_ANALYTICS_4 } from './constants';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+import {
+	combineStores,
+	createValidatedAction,
+} from '../../../googlesitekit/data/utils';
+import { actions as errorStoreActions } from '../../../googlesitekit/data/create-error-store';
+
+const { receiveError, clearError } = errorStoreActions;
+
+const validateAudienceSettings = ( settings ) => {
+	invariant(
+		isPlainObject( settings ),
+		'Audience settings should be an object.'
+	);
+	invariant(
+		Array.isArray( settings.availableAudiences ),
+		'Available audiences should be an array.'
+	);
+	invariant(
+		typeof settings.audienceSegmentationSetupCompletedBy === 'number',
+		'Audience segmentation setup completed by should be an integer.'
```

Comment:
The phrase "Audience segmentation setup completed by should be an integer" doesn't make sense grammatically. Let's update as follows to avoid this:

```suggestion
		isPlainObject( audienceSettings ),
		'audienceSettings should be an object.'
	);
	invariant(
		Array.isArray( audienceSettings.availableAudiences ),
		'availableAudiences should be an array.'
	);
	invariant(
		typeof audienceSettings.audienceSegmentationSetupCompletedBy === 'number',
		'audienceSegmentationSetupCompletedBy should be an integer.'
```